internal/cli/internal/command/ssh/ssh_terminal.go

```diff
@@ -0,0 +1,204 @@
+package ssh
+
+import (
+	"context"
+	"crypto/ed25519"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"time"
+
+	sshCrypt "golang.org/x/crypto/ssh"
+
+	"github.com/briandowns/spinner"
+	"github.com/pkg/errors"
+	"github.com/superfly/flyctl/api"
+	"github.com/superfly/flyctl/helpers"
+	"github.com/superfly/flyctl/internal/client"
+	"github.com/superfly/flyctl/pkg/agent"
+	"github.com/superfly/flyctl/pkg/ssh"
+	"github.com/superfly/flyctl/terminal"
+)
+
+func spin(in, out string) context.CancelFunc {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
we don't use `context.Background`, in the new CLI we carry over the same context so that we can pickup cancellation.